core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -197,7 +197,7 @@ private[spark] object Utils extends Logging {
   /**
    * Get the ClassLoader which loaded Spark.
    */
-  def getSparkClassLoader: ClassLoader = getClass.getClassLoader
+  def getSparkClassLoader: ClassLoader = SparkClassUtils.getSparkClassLoader
```

Comment:
You implement `SparkClassUtils` as a trait, and make `Utils` implement that trait. That will save you from writing a lot of code like this.

Reply:
This is a good idea.